go/libkb/identify3.go

```diff
@@ -202,8 +196,13 @@ func (s *Identify3State) runExpireThread(g *GlobalContext, expireCh <-chan struc
 
 	for {
 		select {
-		case <-shutdownCh:
-			mctx.Debug("identify3State#runExpireThread: exiting on shutdown")
+		case <-mctx.Ctx().Done():
```

Comment:
So I think this only would work if this mctx was shared throughout the code.  The way it is now, it's created in `newIdentify3State` but no one will ever cancel that context.

Unless I'm missing something?

Reply:
this should be the mctx that's created in `newIdentify3State`. my intention (and i think it works this way) is that when you instantiate one of these objects, it creates a cancellable metacontext and passes that around on all its background tasks. it saves the cancel function, and when the object gets shutdown from outside, it executes that cancel function which kills the context on all the background tasks. 

i'll keep looking, but i can't see yet why this shouldn't work...